scripts/update_license_headers.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+import re
+import sys
+
+def update_license_header(file_path):
+    with open(file_path, 'r') as file:
+        content = file.read()
+
+    # Pattern to match copyright lines and Apache license
+    header_pattern = re.compile(
+        r'^(// Copyright.*\n)+'  # Match one or more copyright lines
+        r'(//\s*\n)*'  # Match zero or more empty comment lines
+        r'// Licensed under the Apache License, Version 2\.0.*?'
+        r'limitations under the License\.\n+',
+        re.MULTILINE | re.DOTALL
+    )
+
+    spdx_header = "// SPDX-License-Identifier: Apache-2.0\n"
+
+    new_content, count = header_pattern.subn(
+        lambda m: ''.join(re.findall(r'^// Copyright.*\n', m.group(0), re.MULTILINE)) + spdx_header,
+        content
+    )
+
+    if count > 0:
+        with open(file_path, 'w') as file:
+            file.write(new_content)
+        print(f"Updated {file_path}")
+        return True
+    else:
+        print(f"Skipping {file_path}: No Apache header found")
```

Comment:
suggestion:
* add logic to check for either apache or SPDX header and print a warning when neither one is found for a file
* add an option to the script to run in dry-run mode where we can check first if any of the source files will generate such warnings, so that we can investigate before bulk-applying the change
* do a dry-run yourself to ensure files can be updated as expected

Reply:
Thanks for the suggestions @yurishkuro. I have updated and simplified the script more.

The script currently retains the existing year, but we can modify it to update to the current year if preferred. Please let me know your preference on this.

Reply:
the year should stay as is.

Reply:
right now the script incorrectly warns about missing header on files that already have SPDX
```
Warning: ./pkg/es/errors.go - Could not find expected license header
```

Reply:
Thanks @yurishkuro for the feedback. Updated the code